src/contract/types.rs

```diff
@@ -35,70 +42,140 @@ pub fn encode_document_field_type(
         }
         DocumentFieldType::Integer => {
             let value_as_integer = value.as_integer().ok_or(field_type_match_error)?;
-            let value_as_u64: u64 = value_as_integer
+            let value_as_i64: i64 = value_as_integer
                 .try_into()
                 .map_err(|_| Error::CorruptedData(String::from("expected integer value")))?;
-            Ok(Some(value_as_u64.to_be_bytes().to_vec()))
+
+            encode_integer(value_as_i64)
         }
         DocumentFieldType::Float => {
             let value_as_float = value.as_float().ok_or(field_type_match_error)?;
             Ok(Some(value_as_float.to_be_bytes().to_vec()))
         }
+        DocumentFieldType::ByteArray => {
+            // Byte array could either be raw bytes or encoded as a base64 string
+            if value.is_text() {
+                // Decode base64 string
+                let base64_value = value.as_text().expect("confirmed as text");
+                let value_as_bytes = base64::decode(base64_value).map_err(|_| {
+                    Error::CorruptedData(String::from("bytearray: invalid base64 value"))
+                })?;
+                Ok(Some(value_as_bytes))
+            } else {
+                let value_as_bytes = value.as_bytes().ok_or(field_type_match_error)?;
+                Ok(Some(value_as_bytes.clone()))
+            }
+        }
+        DocumentFieldType::Boolean => {
+            let value_as_boolean = value.as_bool().ok_or(field_type_match_error)?;
+            if value_as_boolean == true {
+                Ok(Some(vec![1]))
+            } else {
+                Ok(Some(vec![0]))
+            }
+        }
+        DocumentFieldType::Date => {
+            let date_string = value.as_text().ok_or(field_type_match_error)?;
+            let date_as_integer: i64 = date_string
+                .parse()
+                .map_err(|_| Error::CorruptedData(String::from("invalid integer string")))?;
+            encode_integer(date_as_integer)
+        }
+        DocumentFieldType::Object => Ok(Some(Vec::new())),
```

Comment:
We don't index objects so we probably should just throw an error

Reply:
True, thanks.